### ISSUE TITLE
config_drive should not be a required option

### ISSUE THREAD

**Issue by diegows** [saltstack/salt#19923]
In the latest version of Salt 2014.7.1, the config_option is required to be set to True to launch instances in Rackspace using salt-cloud. The error without that option is:

```
    ----------
    Error:
        Error creating rax-openstack-controller on OpenStack

        The following exception was thrown by libcloud when trying to run the initial deployment: 400 Bad Request Invalid config_drive provided.
```

My cloud.providers file:

```
rackspace:
  identity_url: 'https://identity.api.rackspacecloud.com/v2.0/tokens'
  compute_name: cloudServersOpenStack
  protocol: ipv4
  compute_region: ORD
  user: diegows
  apikey: XX
  tenant: 123456
  provider: openstack
  ssh_key_file: raxssh.pem
  ssh_key_name: openstack-builder
  script_args: git v2014.7.0
  config_drive: true
```

**Comment by rallytime** [comment]
@diegows I believe this is an intended change with the openstack driver, according to #18096. It sounds like, if you're using user data, then you must set `config_drive` to true. 

Can you verify, @gtmanfred?

**Comment by diegows** [comment]
Ok, but I'm not using user data. And it breaks current configurations.

On Thu, Jan 22, 2015 at 4:20 AM, Nicole Thomas <notifications@github.com>
wrote:

> @diegows <https://github.com/diegows> I believe this is an intended
> change with the openstack driver, according to #18096
> <https://github.com/saltstack/salt/pull/18096>. It sounds like, if you're
> using user data, then you must set config_drive to true.
>
> Can you verify, @gtmanfred <https://github.com/gtmanfred>?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/saltstack/salt/issues/19923#issuecomment-70980796>.
>



-- 
Diego Woitasen
Infrastructure Developer, DevOps Engineer, Linux and Open Source expert
http://www.woitasen.com.ar

**Comment by gtmanfred** [comment]
I will look once I get to work.

**Comment by rallytime** [comment]
@diegows Ah, I was seeing the `user` field and filling that in with `userdata` in my brain. @gtmanfred Thanks for looking! Is the `userdata_file` the same setting you're referring to in your comment/addition about `userdata` needing `config_drive` to be True?

**Comment by gtmanfred** [comment]
It should be ex_config_drive, and it should only be automatically turned on when you add userdata_file

https://github.com/apache/libcloud/blob/trunk/libcloud/compute/drivers/openstack.py#L1178

**Comment by gtmanfred** [comment]
I didn't have time to check it today, cause we were swamped all day, and now I am going home to sleep cause I am sick :) If it doesn't get fixed by the time I get better, I will fix it.

Thanks!
Daniel

**Comment by s0undt3ch** [comment]
Fixed in #22280
